Formula/i/icu4c@75.rb

```diff
@@ -0,0 +1,44 @@
+class Icu4cAT75 < Formula
```

Comment:
Why not make this `icu4c` and create an `icu4c@74`?

Reply:
Still trying out versioning all of `icu4c` (i.e. create `icu4c@75`, make original `icu4c -> icu4c@74`).

Partly to avoid multi-day run. Partly to allow a breadcrumb of old versions that at least won't hard fail deprecated formulae (which we don't test but always get complaints on).

Perhaps future approach would be upstream effort to improve the compatibility mode. It's been brought up a few times but generally assumed to not work yet given attempts in Linux distros.

Reply:
Could avoid a multi-day run with a staging branch.

Reply:
Actually this PR will probably be multi-day for Linux dep test (assuming it triggers as seems to be getting skipped, similar issue as `libgit2`).

The versioning is also to try to reduce churn for deprecated dependents. We are supposed to test and bump these too based on our documentation https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae#deprecation
> Deprecated formulae should continue to be maintained by the Homebrew maintainers so they still build from source and their bottles continue to work (even if unmaintained upstream). If this is not possible, they should be disabled.

Reply:
Going to go ahead with this approach for ICU 75. If previous approach ends up easier, we can switch pretty easily on the next major release (since the unversioned name would be an alias).